### PR TITLE
Fleet View Logs using Loki

### PR DESCRIPTION
**Pull request by swati-kale**
Signed-off-by: Swati Kale <swkale@redhat.com>

## What
MGDSTRM-420 -Develop Fleetwide visualization/dashboards for centralized logging in Grafana

## Why
logs

## How
using loki

### CODE REVIEW

**Comment by david-martin** [comment]
Thanks @swati-kale.
I'm trying to make sense of the flow of the dashboard and how you might use it.
Pasting a screenshot of it here for easier discussion
image

Right now, the dashboard is showing a lot of information at different layeres:

* per namespace aggregation
* per OSD cluster aggregation
* per instance (kafka cluster?)
* per pod
* individual pod logs

This is a great starting point, with lots of reference queries we can work on.

@pb82 I don't mind merging this as is, and iterating on it to improve the flow of the dashboard from a UI/layout point of view, and from a UX point of view for someone trying to reactively debug a problem in a cluster.
Perhaps we write out what that flow should be and identify changes that improve the dashboard.
I'm interested in your thoughts on this.

File: install/resources/monitoring-global/dashboards/logs_dashboard.yaml [hunk 1]
```diff
@@ -0,0 +1,1422 @@
+apiVersion: integreatly.org/v1alpha1
+kind: GrafanaDashboard
+metadata:
+  labels:
+    app: strimzi
+  name: logsDashboard
+  namespace: <namespace>
```
Comment (pb82) [inline install/resources/monitoring-global/dashboards/logs_dashboard.yaml:7]: This should not have a namespace set if there is no Makefile target to replace the placeholder

File: install/resources/monitoring-global/dashboards/logs_dashboard.yaml [hunk 2]
```diff
@@ -0,0 +1,1422 @@
...
+spec:
+  name: logsDashboard.json
+  json: |
+  {
```
Comment (pb82) [inline install/resources/monitoring-global/dashboards/logs_dashboard.yaml:11]: needs to be indented by 1 tab

File: install/resources/monitoring-global/dashboards/logs_dashboard.yaml [hunk 3]
```diff
@@ -0,0 +1,1422 @@
+apiVersion: integreatly.org/v1alpha1
+kind: GrafanaDashboard
+metadata:
+  labels:
+    app: strimzi
+  name: logsDashboard
```
Comment (pb82) [inline install/resources/monitoring-global/dashboards/logs_dashboard.yaml:6]: Should be `logs-dashboard`

**Comment by pb82** [comment]
the piechart plugin needs to be specified, otherwise Grafana will not be able to find it:

```
  plugins:
    - name: "grafana-piechart-panel"
      version: "1.5.0"
```

File: install/resources/monitoring-global/dashboards/logs_dashboard.yaml [hunk 4]
```diff
@@ -0,0 +1,1422 @@
...
+      "pointradius": 2,
+      "points": false,
+      "renderer": "flot",
+      "seriesOverrides": [],
+      "spaceLength": 10,
+      "stack": false,
+      "steppedLine": false,
+      "targets": [
+        {
+          "expr": "sum(count_over_time({strimzi_io_kind=~\"$kind\",namespace=~\"$namespace\",container_name=~\"$container_name\",statefulset_kubernetes_io_pod_name=~\"$pods\",cluster_id=~\"$cluster_id\" ,app_kubernetes_io_name=\"kafka\"}  |~ \"ERROR\"  [1m])) by (tenant)",
```
Comment (pb82) [inline install/resources/monitoring-global/dashboards/logs_dashboard.yaml:979]: why is it `by (tenant)` here when the warnings panel is `by (namespace)`?

Comment (swati-kale) [reply install/resources/monitoring-global/dashboards/logs_dashboard.yaml:979]: My bad, i moved dashboard from other env where I was seeing tenants, will chnge it.

**Review by pb82 — APPROVED** [review]

**Comment by redmikhail** [comment]
@swati-kale I wonder if we can do global replace of `statefulset_kubernetes_io_pod_name` to `instance` and change `app_kubernetes_io_name` to use `container_name` instead like : ``` "sum(count_over_time({strimzi_io_kind=~\"$kind\",namespace=~\"$namespace\",container_name=~\"kafka\",instance=~\"$pods\",cluster_id=~\"$cluster_id\"}  |~ \"WARN | Warning\"  [1m])) by (namespace)"``` Or we can merge it and update subsequently ....

**Comment by redmikhail** [comment]
I have made the changes in the same pull request

**Pull request merged**